Apps/W1/File - Azure BLOB Storage Connector/app/src/BlobStorageConnectorImpl.Codeunit.al

```diff
@@ -0,0 +1,414 @@
+        if not ABSOperationResponse.IsSuccessful() then
+            Error(ABSOperationResponse.GetError());
+
+        exit(not ABSContainerContent.IsEmpty());
+    end;
+
+    /// <summary>
+    /// Deletes a file exists on the provided account.
+    /// </summary>
+    /// <param name="AccountId">The file account ID which is used to send out the file.</param>
+    /// <param name="Path">The file path inside the file account.</param>
+    procedure DeleteFile(AccountId: Guid; Path: Text)
+    var
+        ABSBlobClient: Codeunit "ABS Blob Client";
+        ABSOperationResponse: Codeunit "ABS Operation Response";
+    begin
+        InitBlobClient(AccountId, ABSBlobClient);
+        ABSOperationResponse := ABSBlobClient.DeleteBlob(Path);
+
+        if ABSOperationResponse.IsSuccessful() then
+            exit;
+
+        Error(ABSOperationResponse.GetError());
+    end;
+
+    /// <summary>
+    /// Gets a List of Directories stored on the provided account.
+    /// </summary>
+    /// <param name="AccountId">The file account ID which is used to get the file.</param>
+    /// <param name="Path">The file path to list.</param>
+    /// <param name="Files">A list with all directories stored in the path.</param>
+    procedure ListDirectories(AccountId: Guid; Path: Text; var FileAccountContent: Record "File Account Content" temporary)
+    var
+        ABSContainerContent: Record "ABS Container Content";
+        ABSBlobClient: Codeunit "ABS Blob Client";
+        ABSOperationResponse: Codeunit "ABS Operation Response";
+        ABSOptionalParameters: Codeunit "ABS Optional Parameters";
+    begin
+        InitBlobClient(AccountId, ABSBlobClient);
+
+        if (Path <> '') and not Path.EndsWith('/') then
+            Path += '/';
+
+        ABSOptionalParameters.Prefix(Path);
+        ABSOperationResponse := ABSBlobClient.ListBlobs(ABSContainerContent, ABSOptionalParameters);
+
+        if not ABSOperationResponse.IsSuccessful() then
+            Error(ABSOperationResponse.GetError());
+
+        ABSContainerContent.SetRange("Parent Directory", Path);
+        ABSContainerContent.SetRange("Blob Type", '');
+        if not ABSContainerContent.FindSet() then
+            exit;
+
+        repeat
+            FileAccountContent.Init();
+            FileAccountContent.Name := ABSContainerContent.Name;
+            FileAccountContent.Type := FileAccountContent.Type::Directory;
+            FileAccountContent."Parent Directory" := ABSContainerContent."Parent Directory";
+            FileAccountContent.Insert();
+        until ABSContainerContent.Next() = 0;
+    end;
+
+    /// <summary>
+    /// Creates a directory on the provided account.
+    /// </summary>
+    /// <param name="AccountId">The file account ID which is used to send out the file.</param>
+    /// <param name="Path">The directory path inside the file account.</param>
+    procedure CreateDirectory(AccountId: Guid; Path: Text)
```

Comment:
After attending @vjekob  session for Testabilit by the use of interface, i think we maybe should split this interface into more interfaces. One General and at least one for directories.